kiosk/kiosk_browser/connection.py

```diff
@@ -0,0 +1,44 @@
+import requests
+import tempfile
+import threading
+import time
+
+check_connection_url = 'http://captive.apple.com/'
+sleep_time = 5
+
+class Connection():
+
+    def __init__(self, set_captive_portal_url):
+        self._is_connected = False
+        self._set_captive_portal_url = set_captive_portal_url
+
+    def start_daemon(self):
+        thread = threading.Thread(target=self._check, args=[])
+        thread.daemon = True
+        thread.start()
+
+    def is_connected(self):
+        return self._is_connected
+
+    def _check(self):
+        # Initial sleep to be sure _set_captive_portal_url is accessible
+        time.sleep(1)
+
+        while True:
+            try:
+                r = requests.get(check_connection_url, allow_redirects = False)
+
+                if r.status_code == 200:
+                    self._is_connected = True
+                    self._set_captive_portal_url('')
+                elif r.status_code in [301, 302, 303, 307]:
+                    self._is_connected = False
+                    self._set_captive_portal_url(r.headers['Location'])
+                else:
```

Comment:
Now that this branch does not trigger the captive portal, It’s ok to reset the url.

We could also show a message to the user in this case, something like “There is some trouble with the connection right now”. But it could be a false positive in case there is something unpredictable going on the captive url. Or, the application itself could already show an error message if the connection is really down.

Now, in case the user is behind a captive portal, and this captive portal use a method not handled here, it could be interesting.

With pros and cons, I would keep it as it is.

Reply:
I agree, let's go with this implementation and observe.